API/Services/ArchiveService.cs

```diff
@@ -57,12 +57,8 @@ public virtual ArchiveLibrary CanOpen(string archivePath)
         {
             if (string.IsNullOrEmpty(archivePath) || !(File.Exists(archivePath) && Parser.Parser.IsArchive(archivePath) || Parser.Parser.IsEpub(archivePath))) return ArchiveLibrary.NotSupported;
 
-            try
-            {
-                using var a2 = ZipFile.OpenRead(archivePath);
-                return ArchiveLibrary.Default;
-            }
-            catch (Exception)
+            var ext = _directoryService.FileSystem.Path.GetExtension(archivePath);
+            if (ext == ".cbr" || ext == ".rar")
```

Comment:
Change this to if (ext == ".cbr" || ext == ".rar") return ArchiveLibrary.SharpCompress

Reply:
Also make sure you do a .toLower() (actually .toUpper() would be better) because some files are .RAR

Reply:
Resolve in commit 731d078